src/loaders/genome_collection/compute_genome_taxa_count.py

```diff
@@ -0,0 +1,173 @@
+import argparse
+import json
+import os
+from collections import defaultdict
+
+import src.common.storage.collection_and_field_names as names
+import src.loaders.common.loader_common_names as loader_common_names
+from src.common.gtdb_lineage import (
+    parse_gtdb_lineage_string,
+    GTDB_RANK_ABBREV_TO_FULL_NAME,
+)
+from src.common.hash import md5_string
+from src.loaders.common.loader_helper import convert_to_json
+from src.service.data_products import taxa_count
+
+"""
+PROTOTYPE - Prepare genome taxa count data and identical ranks in JSON format for arango import.
+
+usage: compute_genome_taxa_count.py [-h] --load_ver LOAD_VER [--kbase_collection KBASE_COLLECTION] [-o OUTPUT]
+                                    [--input_source {GTDB,genome_attributes}]
+                                    load_files [load_files ...]
+
+options:
+  -h, --help            show this help message and exit
+
+required named arguments:
+  load_files            Files containing genome taxonomy (e.g. ar53_taxonomy_r207.tsv, computed_genome_attribs.json)
+  --load_ver LOAD_VER   KBase load version (e.g. r207.kbase.1).
+
+optional arguments:
+  --kbase_collection KBASE_COLLECTION
+                        KBase collection identifier name (default: GTDB).
+  -o OUTPUT, --output OUTPUT
+                        Output JSON file path.
+  --input_source {GTDB,genome_attributes}
+                        Input file source
+ 
+e.g. compute_genome_taxa_count.py bac120_taxonomy_r207.tsv ar53_taxonomy_r207.tsv --load_version 207
+     compute_genome_taxa_count.py bac120_taxonomy_r207.tsv ar53_taxonomy_r207.tsv --load_version 207 --kbase_collection GTDB
+     compute_genome_taxa_count.py bac120_taxonomy_r207.tsv ar53_taxonomy_r207.tsv --load_version 207 --kbase_collection GTDB --output  gtdb_taxa_counts.json
+"""
+
+# Default result file name for genome taxa count data and identical ranks for arango import
+# Collection and load version information will be prepended to this file name.
+GENOME_TAXA_COUNT_FILE = "parsed_genome_taxa_counts.json"
+
+# The source of the input file containing genome taxonomy information to be parsed
+# GTDB - taxonomy file downloaded directly from the GTDB website, such as 'bac120_taxonomy_r207.tsv'
+# genome_attributes - genome attributes file created by running 'parse_computed_genome_attribs.py' script.
+VALID_SOURCE = ['GTDB', 'genome_attributes']
+
+# taxonomy attribute name from computed genome attributes
+TAXA_ATTRI_NAME = 'classification'
```

Comment:
Hold on - this should be `gtdb_taxonomy`. This makes me think we may have miscommunicated about what exactly this code is supposed to take as input, so I'm going to hold off on reviewing the rest of this script until we sort that out.

My understanding is that it should be working on the JSONL files produced by the parsing script, but I don't think that's what's happening here

Reply:
It turns out that the code is reading from the right file but the gtdb lineage key is different, which is a problem, but we'll deal with that later. https://kbase-jira.atlassian.net/browse/RE2022-99

Reply:
Continuing on with review below...

Reply:
never mind, looks fine otherwise. It's mostly identical to the old script other than input.

However - the difference between the keys between the output of the GTDB metadata parser and the compute -> parser scripts means this can't run on the output of the GTDB metadata parser. If the keys are the same, than we can get rid of the direct support for the GTDB metadata file and just run it on the parser output